hs-abci-sdk/src/Tendermint/SDK/Modules/Validators/Types.hs

```diff
@@ -0,0 +1,55 @@
+module Tendermint.SDK.Modules.Validators.Types where
+
+import           Control.Lens                           (iso)
+import qualified Data.Aeson                             as A
+import           Data.ByteString                        (ByteString)
+import           Data.ByteString.Lazy                   (toStrict)
+import           Data.Maybe                             (fromJust)
+import           Data.Set                               (Set)
+import           Data.Word                              (Word64)
+import           GHC.Generics                           (Generic)
+import           Network.ABCI.Types.Messages.FieldTypes (PubKey)
+import           Tendermint.SDK.BaseApp                 (RawKey (..))
+import           Tendermint.SDK.Codec                   (HasCodec (..))
+
+
+data ValidatorsNameSpace
+
+type ValidatorsName = "validators"
+
+
+updatesListKey :: ByteString
+updatesListKey = "updatesList"
+
+validatorsMapKey :: ByteString
+validatorsMapKey = "validatorsMap"
+
+validatorsKeySetKey :: ByteString
+validatorsKeySetKey = "validatorsKeySet"
+
+
+data ValidatorUpdate = ValidatorUpdate
+    { key   :: PubKey_
+    , power :: Word64
+    }
+    deriving Generic
+instance A.ToJSON ValidatorUpdate
+instance A.FromJSON ValidatorUpdate
+instance HasCodec ValidatorUpdate where
```

Comment:
For the state objects we would like to use a `.proto` file to describe the objects, see for example how the `Auth` module is done. https://github.com/f-o-a-m/kepler/tree/master/hs-abci-sdk/protos/modules

There are several reasons why we do this, you can see this discussion if you're interested: https://github.com/cosmos/cosmos-rust/issues/14#issuecomment-748174290